File: android/demos/SwapiAPIExample/app/src/main/java/com/example/swapiapiexample/data/model/CharacterResponse.kt
package com.example.swapiapiexample.data.model

import com.google.gson.annotations.SerializedName

/**
 * API response wrapper for paginated character results from SWAPI
 */
data class CharacterResponse(
    @SerializedName("count")
    val count: Int,

    @SerializedName("next")
    val next: String?,

    @SerializedName("previous")
    val previous: String?,

    @SerializedName("results")
    val results: List<Character>
)

File: ios/demos/starwars-api/StarwarsAPI/StarwarsAPI/CharacterResponse.swift
//
//  CharacterResponse.swift
//  StarwarsAPI
//
//  Created by Nick Todd on 24/10/2025.
//

import Foundation

/// Represents the API response wrapper from SWAPI's /people/ endpoint
struct CharacterResponse: Codable {
    let count: Int
    let next: String?
    let previous: String?
    let results: [Character]
}
